src/main/scala/com/qubole/sparklens/common/AppContext.scala

```diff
@@ -44,6 +44,35 @@ case class AppContext(appInfo:        ApplicationInfo,
                      x._2.endTime <= endTime),
       stageIDToJobID)
   }
+}
+
+object AppContext {
+
+  def getMaxConcurrent[Span <: TimeSpan](map: mutable.HashMap[String, Span]): Long = {
+
+    // sort all start and end times on basis of timing
+    val sorted = map.values.flatMap(timeSpan => {
+      val correctedEndTime = if (timeSpan.endTime == 0) {
+        System.currentTimeMillis()
```

Comment:
We might need to revisit how to deal with missing end time when we produce report from JSON.

Reply:
Yes this is being taken care at `AppContext.getMaxConcurrent()` function,